CommunityFridgeMapApi/dependencies/python/db.py

```diff
@@ -220,6 +161,18 @@ def is_valid_display_name(self) -> bool:
         # A valid display_name is alphanumric and can contain spaces
         username = self.display_name.lower().replace(" ", "")
         return username.isalnum()
+    
+    @staticmethod
+    def is_valid_username(fridge_username) -> tuple[bool, str]:
```

Comment:
Why is a tuple being returned here?

Reply:
A tuple was returned in the Tag class in the corresponding function (is_valid_tag_name). It seems consistent to me on that account.

Reply:
bool: is valid username or is not valid username
str: error message if there is one

The main reason is to return a different error message depending on the reason for invalidity

Reply:
Using a tuple then is redundant imo. The message would automatically determine the bool, so we should wrap this logic in an enum consisting of error messages. Since this change would be project wide, let's create a ticket for this and pick it later?

Reply:
I agree that it's redundant but I think its a code readability thing for me. 

With the implementation you are suggesting the dev would do?

```
if "message" in "username_error_message":
```

Reply:
Something like
```python
class Result:
    static InvalidLengthMessage = "length..."
    static SuccessMessage = "success..."
```
Then use it like
```python
def func() -> Result:
    # do something

if func() == Result.SuccessMessage:
    # something
else:
    # handle error with string
```
This is just an initial idea. Other languages usually have a `Result<Data, Error>` type that encapsulates enums, we could do something similar for Python here.

Reply:
I think I still prefer doing a bool check over doing a string comparison. It feels more natural to me. This doesn't have to be in a tuple though, it can be in a Result class like you suggested. 

I do like the idea of the validation functions being responsible for returning the error messages and having that be consistent throughout the code.

I have to look into the pros and cons of having all the error messages within a Result class. Most of the time the error messages are one-offs and so I could see the Result class being filled with dozens/hundreds of error messages as the code base grows. That being said we could probably abstract out some of the error messaging. 

Some more thought should go into this, I agree lets leave it for another ticket